### ISSUE TITLE
"Discovered" licenses from NOTICE file showing up in "Declared" field - Google Mavens

### ISSUE THREAD

**Issue by ariel11** [clearlydefined/crawler#583]
There is a bug with Google Maven packages (maybe other package types too?) where a notices file license data (which should be "discovered" licenses) are erroneously included in the "declared" field.  

Example: https://clearlydefined.io/definitions/maven/mavengoogle/com.google.android.gms/play-services-location/21.2.0. 

All the info from the third party notices file is being erroneously included in the "declared" field - those should be "discovered" licenses. In this case, the "declared" would be "OTHER" since there's not a SPDX ID for the "Android Software Development Kit License."

<img width="792" alt="image" src="https://github.com/clearlydefined/crawler/assets/17935946/1ae0f602-c94f-497f-8a16-dd46e203d5d4">

**Comment by ariel11** [comment]
@capfei - I thought we had an open Issue about this already but I couldn't find it?

@elrayle - FYI, this is a significant pain point I am hoping to get on your radar. Happy to chat more.

**Comment by qtomlinson** [comment]
@ariel11 @elrayle @capfei Yes, there was a previous [similar issue](https://github.com/clearlydefined/crawler/issues/506) on this.   Similar to [my comment on that issue](https://github.com/clearlydefined/crawler/issues/506#issuecomment-1811586558):  in the ScanCode v30 result, there is no package level license information, so license information for top level files is used to derive the declared license. is_license_text is true for third_party_licenses.txt in ScanCode v30 result, and therefore the licenses matched are used as the declared license.  We can rerun this case after [ScanCode upgrage PR](https://github.com/clearlydefined/service/pull/1056) is merged.

**Comment by ariel11** [comment]
@qtomlinson - I ran into this same issue for PyPI - all the NOTICE file attributions are ending up in the declared field. NOTICE file attributions should always only be "discovered" licenses. 

https://clearlydefined.io/definitions/pypi/pypi/-/azure-ai-ml/1.21.1

![Image](https://github.com/user-attachments/assets/a4ae74bb-2a68-45be-b7a5-ffaca23da111)

On the "Files" tab:

<img width="731" alt="Image" src="https://github.com/user-attachments/assets/c7767e19-25d8-41d5-8431-689795e1e631" />

**Comment by ariel11** [comment]
@qtomlinson - Hi, it looks like the Scancode update PR referenced above was merged June 26, 2024. We continue to see instances of the notice file licenses appearing in the declared field. 

I don't know what other ClearlyDefined users would prefer, but I would actually prefer for the ClearlyDefined tooling to put NOASSERTION in the declared field _vs_. copying all the "discovered" licenses from a notices file into the declared field when it cannot tell what the top level license should be. This causes a great deal of noise. 

Latest NuGet example - the top level license for this NuGet is MIT. 

![Image](https://github.com/user-attachments/assets/30d9baf6-79bf-49f5-bfa9-3c9fd6093256)

**Comment by JustinGOSSES** [comment]
This issue has been reported to me from 4+ parties in past week as well. 

Packages where it seems ClearlyDefined is reporting incorrect licenses sourced from NOTICE files include: 
- https://clearlydefined.io/?sort=releaseDate&sortDesc=true&name=Microsoft.Azure.Cosmos
- [Microsoft.AzureStreamJsonRpc](https://clearlydefined.io/?sort=releaseDate&sortDesc=true&name=StreamJsonRpc)
- https://clearlydefined.io/?sort=releaseDate&sortDesc=true&name=1ds-core-js
- https://clearlydefined.io/?sort=releaseDate&sortDesc=true&name=1ds-post-js

Please let me know how I can help debug.

**Comment by jamesiri** [comment]
I was able to track this down to an issue with the version of scancode we are using. I am working on getting an update but it is slow going since I have to run a scan repeatedly and I can't seem to speed it up at all so it takes 30 minutes per run. this is my main goal this week.
